app/assets/javascripts/components/models/Reaction.js

```diff
@@ -135,6 +135,14 @@ export default class Reaction extends Element {
         break
     }
 
+    // If userText is number only, treat as normal temperature value
+    if (/^\-?\d*\.{0,1}\d{0,2}$/.test(temperature.userText)) {
+      temperature.userText =
+        convertFunc(newUnit, temperature.userText).toFixed(2)
+
+      return temperature
+    }
+
     temperature.data.forEach(function(data, index, theArray) {
       theArray[index].value = convertFunc(newUnit, data.value).toFixed(2)
```

Comment:
if you Fixed(2) every time you change unit, you loose precision. displayed value should be Fixed(2) but stored value should have more